src/app/ssh-keys/ssh-key-sidebar/ssh-key-sidebar.component.ts

```diff
@@ -15,24 +15,31 @@ import { EntityDoesNotExistError } from '../../shared/components/sidebar/entity-
 })
 export class SshKeySidebarComponent extends SidebarComponent<SSHKeyPair> implements OnInit {
   public description: string;
+  public account: string;
 
-  constructor(
-    protected entityService: SSHKeyPairService,
-    protected notificationService: NotificationService,
-    protected route: ActivatedRoute,
-    protected router: Router,
-    protected userTagService: UserTagService
-  ) {
+  constructor(protected entityService: SSHKeyPairService,
+              protected notificationService: NotificationService,
+              protected route: ActivatedRoute,
+              protected router: Router,
+              protected userTagService: UserTagService) {
     super(entityService, notificationService, route, router);
   }
 
+  public ngOnInit() {
+    this.getEntityAccount();
+
+    this.pluckId()
+      .switchMap(id => this.loadEntity(id, this.account))
+      .subscribe(entity => this.entity = entity);
+  }
+
   public onDescriptionChange(description: string): void {
     this.description = description;
     this.userTagService.setSshKeyDescription(this.entity, this.description).subscribe();
   }
 
-  protected loadEntity(name: string): Observable<SSHKeyPair> {
-    return this.entityService.getByName(name)
+  protected loadEntity(name: string, account?: string): Observable<SSHKeyPair> {
```

Comment:
can we leave this method with a single parameter and fetch account from stateParams here?